lesson 4 - done

AoifeSidhe opened the pull request:
task - homework
notask - мой фаил для наглядного примера мне лично.

AoifeSidhe committed:
Add lesson 4 homework files

This commit adds three files for lesson 4 homework. They include task.py, nontask.py, and a Readme.md description file. In task.py, various operations are demonstrated on python data structures including lists, tuples, sets and dictionaries. In nontask.py, a nested dictionary creation and operations on it are demonstrated. Feel free to skip it! The Readme.md file gives a detailed explanation of the contents and usage of both python script files.

AoifeSidhe committed:
Refactor print statements in lesson_4/task.py

AoifeSidhe committed:
Damn pylinter =)

AoifeSidhe committed:
pylinterfix - C0305: Trailing newlines (trailing-newlines)

AoifeSidhe commented:
![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/694e59c5-0906-415a-8ee2-2e5d81b30a0d)

eugene-okulik commented on homework/aoife_sidhe/lesson_4/task.py:
```diff
@@ -0,0 +1,140 @@
+display_dict()  # default dict
+print()
+print("Adding new key-value pair for prices_dict - i am a tuple")
+my_dict['prices_dict']['i am a tuple'] = 100  # add a new item
```
В задании было

> добавьте элемент с ключом ('i am a tuple',) и любым значением

ты выполнила так, как если бы там было 

> добавьте элемент с ключом 'i am a tuple' и любым значением

AoifeSidhe replied:
Я думала, что  скобки это часть форматирования задания =)
Хотя запятую да, в упор не видела
Поправила. 
Задание на то, что надо правильно кавычки включать?

И так пройдёт:
```
print("Adding new key-value pair for prices_dict - i am a tuple")
my_dict['prices_dict']['(\'i am a tuple\',)'] = 100
```

А так нет:
`my_dict['prices_dict']['('i am a tuple',)'] = 100  # add a new item`

![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/4461ffaa-ed74-4c66-972e-ed05d51bcdf8)

П.с. У пайчарма беда - он не обновдяет ошибку при исправлении. То есть я сделала опечатку - он подсветил. Я поправила - показывает по прежнему ошибку со старым значением.
Помогает - сtrl-x/ctrl-v =)

В картинках:

Ошибка:
![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/49ef0361-14b8-46c5-bf2f-af724ae9027d)

Все равно ошибка:
![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/bcb052bb-dc53-4392-bad2-5788782cb43a)

eugene-okulik reviewed (changes requested):
Мне, видимо, нужно привыкнуть к тому, что ты будешь выполнять еще и задания, которые сама придумала )))  
В целом, по заданию всё выполнено, но проверку на внимательность не прошла ))

AoifeSidhe committed:
Modify task.py

In the task.py file, corrected the code for adding a new key-value pair in 'prices_dict'.

eugene-okulik commented on homework/aoife_sidhe/lesson_4/task.py:
```diff
@@ -0,0 +1,140 @@
+display_dict()  # default dict
+print()
+print("Adding new key-value pair for prices_dict - i am a tuple")
+my_dict['prices_dict']['(\'i am a tuple\',)'] = 100  # add a new key-value pair
```
На что похоже `('i am a tuple', )`, на какой тип данных? И что говорит текст внутри нее?  
Сейчас тот ключ, который ты добавила, это строка `'(\'i am a tuple\',)'`. И, кстати, можно играться разными кавычками, чтобы не приходилось расставлять эти слеши: `"('i am a tuple',)"`

AoifeSidhe replied:
tuple похож на tuple
но если быть душным, то 
> добавьте элемент с ключом ('i am a tuple',) и любым значениемдобавьте элемент с ключом ('i am a tuple',) и любым значением

не отрицает что этот элемент может быть хоть чем угодно =)
Но по факту - надо было добавить tuple в словарь?)

Переделаю =)
Правда тапл потянул за собой еще пару изменений, но их тоже внесу.

AoifeSidhe replied:
![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/9aee851a-c79d-4fb5-8eaa-40aad79ff9a9)

and

![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/32436716-63b7-42d1-ac47-4726c0b83a79)

AoifeSidhe replied:
Сделала файл clear_task чтобы можно было посмотреть код без моих подробных комментариев для самой себя.
И simple_task - где почти без лишнего.
Все остальные файлы - это мне на будущее, чтобы в голове уложить информацию.

Код теперь даже показывает, что наш тапл - это тапл, а не какой-то там стринг или, ужас какой, int.
Прикрутила простое меню иначе проверять этот ужас в терминале было невыносимо.
Меню, кстати, на словаре работает.

![image](https://github.com/eugene-okulik/AoifeSidhe/assets/127910230/98ea8483-984e-428d-bc86-f7b65f14e575)

AoifeSidhe committed:
Fix and several files for my personal study

personal: tutorial notebooks add notes.txt
for review: clear_task.py to view code without my too detailed  comments

fix tuple problems

AoifeSidhe committed:
Update notes.txt and extend 'sets.ipynb' notebook

Extended lesson_4/notes.txt with code snippets and explanations on various Python concepts and functionalities.
Additionally, updated and enriched 'sets.ipynb' notebook with more information about set operations like discard, difference, and symmetric_difference methods.
Presented some practical examples of using Set in Python.

AoifeSidhe committed:
fix for tuple ready for review

Add ASCII art to file.txt
This piece of ASCII art depicts a cat!

AoifeSidhe committed:
Move Aoife's clear_task.py out of .ipynb_checkpoints

The clear_task.py file was mistakenly placed in the .ipynb_checkpoints directory. This commit moves it to the appropriate directory, which is the main lesson_4 directory.

AoifeSidhe committed:
Refactor print statements and remove unused function call - homework 4 is done.

Damn f-strings!

AoifeSidhe committed:
Refactor print statements in lesson 4 homework.

AoifeSidhe committed:
pylinter - to run a check

AoifeSidhe committed:
R0912 - trying to solve

AoifeSidhe committed:
R0912 - trying to solve

 homework/aoife_sidhe/lesson_4/task.py:126:0: R0912: Too many branches (18/12) (too-many-branches)
homework/aoife_sidhe/lesson_4/task.py:126:0: R0915: Too many statements (68/50) (too-many-statements)

AoifeSidhe committed:
Update lesson 4 notes and tasks. My own goal to pass the pylinter.

Updated notes with additional information on error handling, built-in functions and tuple handling in Python. Made modifications to tasks and exercises in .ipynb and .py files, focusing on tuple, dictionaries, sets, and list operations. Also performed refactorings for code cleanliness and adherence to best practices.

AoifeSidhe committed:
Homework4 is done. W292 (C0304) fix

./homework/aoife_sidhe/lesson_4/task.py:379:74: W292 no newline at end of file
homework/aoife_sidhe/lesson_4/task.py:379:0: C0304: Final newline missing (missing-final-newline)

AoifeSidhe committed:
add simple_task.py

- add most simple task where we solve homework as easy as we can. But with several checks

- fix for script logic (it was broken for type task_e_way_b() )

AoifeSidhe committed:
E302 fix

AoifeSidhe committed:
Module simple_task - linter fix

homework/aoife_sidhe/lesson_4/simple_task.py:34:0: R1707: Disallow trailing comma tuple (trailing-comma-tuple)
solved
homework/aoife_sidhe/lesson_4/simple_task.py:20:0: C0103: Constant name "last_element_tuple" doesn't conform to UPPER_CASE naming style (invalid-name)
solved
homework/aoife_sidhe/lesson_4/simple_task.py:49:0: C0103: Constant name "key_name" doesn't conform to UPPER_CASE naming style (invalid-name)
solved
homework/aoife_sidhe/lesson_4/simple_task.py:51:0: C0103: Constant name "value_type" doesn't conform to UPPER_CASE naming style (invalid-name)
solved
homework/aoife_sidhe/lesson_4/simple_task.py:70:0: C0206: Consider iterating with .items() (consider-using-dict-items)
ignored
homework/aoife_sidhe/lesson_4/simple_task.py:70:11: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)
solved
homework/aoife_sidhe/lesson_4/simple_task.py:79:11: C0201: Consider iterating the dictionary directly instead of calling .keys() (consider-iterating-dictionary)
solved

AoifeSidhe committed:
Refactor dictionary iteration in simple_task.py

AoifeSidhe committed:
R1701:

R1701: Consider merging these isinstance calls to isinstance(dict_value, (list, set)) (consider-merging-isinstance)

AoifeSidhe committed:
Homework 4 is done twice.

eugene-okulik reviewed (approved):
Ничего себе какая куча файлов!  
Файла с чистым выполнением домашнего задания я так и не нашел, даже в этом simple_task есть куча кода, который я не очень понимаю что делает.  
Если ты хочешь экспериментировать, то создай отдельную папку в папке с заданием, назови это папку "сюда не смотри" и складывай туда эксперименты, а то не понятно где мне нужно проверить домашнее задание. И в самом выполнении сложно среди экспериментов найти то, что было в задании.  
А с котрежем - я хотел обратить внимание на то, что кортеж может быть ключом словаря и заодно обратить внимание на то как оформляется кортеж с одним элементом.

AoifeSidhe commented:
>  этом simple_task есть куча кода, который я не очень понимаю что делает
Там одна кучка - это проверка на то tuple на то, что это именно он. Потому что я в этом моменте как раз хотела дополнительную проверку, что оно именно так, а я не ошиблась.
А вторая кучка - просто красиво печатает это все в терминал =)
Потому что простой принт выдает какую-то не очень читаемую строку. 

А часто ли tuple используется как ключ?

Создам тогда внутри паку - experiments и буду скаладывать свою доп работу туда.

eugene-okulik commented:
> > этом simple_task есть куча кода, который я не очень понимаю что делает
> > Там одна кучка - это проверка на то tuple на то, что это именно он. Потому что я в этом моменте как раз хотела дополнительную проверку, что оно именно так, а я не ошиблась.
> > А вторая кучка - просто красиво печатает это все в терминал =)
> > Потому что простой принт выдает какую-то не очень читаемую строку.
> 
> А часто ли tuple используется как ключ?
> 
> Создам тогда внутри паку - experiments и буду скаладывать свою доп работу туда.

Не, не часто. Но когда это нужно, то неочевидно, что такое можно делать. Поэтому я вот и обращаю внимание, чтобы в голове это осталось  
А то для меня это было чуть ли не открытием однажды, хотя я уже тогда и не новичком был.
